lib/shoryuken/message.rb

```diff
@@ -1,11 +1,22 @@
 module Shoryuken
   class Message
-    attr_accessor :client, :queue_url, :data
+    attr_accessor :client, :queue_url, :queue_name, :data
 
-    def initialize(client, queue_url, data)
-      self.client = client
-      self.queue_url = queue_url
-      self.data = data
+    def initialize(client, queue, data)
+      client = client
+
+      if queue.is_a?(Shoryuken::Queue)
+        queue_url = queue.url
+        queue_name = queue.name
+      else
+        # TODO: Remove next major release
+        Shoryuken.loggger.warn do
+          "[DEPRECATION] Passing a queue url into Shoryuken::Message is deprecated, please pass the queue itself"
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.